plugin/core/configurations.py

```diff
@@ -64,10 +63,18 @@ def is_supported(self, view: Any) -> bool:
         return any(self.match_view(view))
 
     def update(self) -> None:
-        self.all = self._create_window_configs()
-        for config in self.all:
-            if config.name in self._temp_disabled_configs:
-                config.enabled = False
+        project_settings = (self._window.project_data() or {}).get("settings", {}).get("LSP", {})
+        self.all.clear()
+        for name in self._global_configs.keys():
+            config = self._global_configs[name]
+            overrides = project_settings.get(config.name)
+            if isinstance(overrides, dict):
+                debug('applying .sublime-project override for', config.name)
+                config = config.update(overrides)
+            self.all[name] = config
+        for name in self._temp_disabled_configs:
+            self.all[name].enabled = False
+        self._window.run_command("lsp_recheck_sessions")
```

Comment:
This breaks some things.

This command triggers `restart_sessions_async` from a timeout.
But at that point the sessions might have already started initializing so that command, depending on timing, will interrupt that.

Reply:
Well, I'm spamming CTRL+S in LSP.sublime-settings and it seems to keep working. I think I accounted for this in `WindowManager.register_listener_async`: it checks if a session is initializing via `self._new_session`. If so, we do nothing, otherwise we do things.

In `register_listener_async`:
```
        self._pending_listeners.appendleft(listener)
        if self._new_listener is None:
            self._dequeue_listener_async()
```
And then later, in `on_post_initialize`:
```
    def on_post_initialize(self, session: Session) -> None:
        sublime.set_timeout_async(self._dequeue_listener_async)
```
It's kind of a complex set up. This would be massively simplified by using `async` functions, or possibly promises.

Reply:
I can reproduce by switching projects using "quick switch project".
I'm switching from empty project to one that uses `gopls` with active file triggering the server and the server doesn't start after switching to gopls project.

Reply:
Actually this seems to be due to the code that updates workspace folders being gone:

In `on_load_project_async` there was:
```py
        workspace_folders = self._workspace.update()
        for session in self._sessions:
            session.update_folders(workspace_folders)
```

Reply:
This line:
```
self._workspace.update()
```
is needed to update the folders. Otherwise we run into the "do or do not attach servers for files outside workspace" discussion.